x-pack/plugins/observability_solution/infra/server/routes/infra/lib/host/get_all_hosts.ts

```diff
@@ -5,43 +5,63 @@
  * 2.0.
  */
 
-import { estypes } from '@elastic/elasticsearch';
-import { lastValueFrom } from 'rxjs';
-import { ESSearchRequest } from '@kbn/es-types';
-import { InfraStaticSourceConfiguration } from '../../../../lib/sources';
-import { decodeOrThrow } from '../../../../../common/runtime_types';
-import { GetInfraMetricsRequestBodyPayload } from '../../../../../common/http_api/infra';
-import { BUCKET_KEY, MAX_SIZE, METADATA_AGGREGATION } from '../constants';
-import {
-  GetHostsArgs,
-  HostsMetricsSearchAggregationResponse,
-  HostsMetricsSearchAggregationResponseRT,
-} from '../types';
-import { createFilters, getInventoryModelAggregations, runQuery } from '../helpers/query';
+import type {
+  GetInfraMetricsRequestBodyPayload,
+  InfraAssetMetadataType,
+} from '../../../../../common/http_api/infra';
+import { BUCKET_KEY, MAX_SIZE, METADATA_AGGREGATION_NAME } from '../constants';
+import type { GetHostsArgs } from '../types';
+import { createFilters, getInventoryModelAggregations } from '../helpers/query';
+import { BasicMetricValueRT } from '../../../../lib/metrics/types';
 
 export const getAllHosts = async (
-  { searchClient, sourceConfig, params }: GetHostsArgs,
+  { infraMetricsClient, params }: GetHostsArgs,
   hostNamesShortList: string[] = []
-): Promise<HostsMetricsSearchAggregationResponse> => {
-  const query = createQuery(params, sourceConfig, hostNamesShortList);
-  return lastValueFrom(
-    runQuery(searchClient, query, decodeOrThrow(HostsMetricsSearchAggregationResponseRT))
```

Comment:
Are we no longer interested in decoding the response here?  Have we lost that ability with the type inference?

Reply:
I don't think we need it anymore. It's too much effort to decode the elasticsearch response if we consider the io-ts types we need to create for that. The `InferSearchResponseOf` type used in the client will infer the correct type. The endpoint response is still decoded thought.